### PR TITLE
fix: exceptionHandler may cause HTTPException: Unknown HTTP status code

### LINKED ISSUE #6227: Bug: Fatal error:  Uncaught CodeIgniter\HTTP\Exceptions\HTTPException: Unknown HTTP status code provided with no message: 167
### PHP Version

8.1

### CodeIgniter4 Version

4

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

_No response_

### What happened?

ehm id write an api and this comes:
```
<b>Fatal error</b>:  Uncaught CodeIgniter\HTTP\Exceptions\HTTPException: Unknown HTTP status code provided with no message: 167 in /var/www/vhosts/blackserv.de/api.blackserv.de/system/HTTP/ResponseTrait.php: 158
Stack trace:
#0 /var/www/vhosts/blackserv.de/api.blackserv.de/system/HTTP/ResponseTrait.php(158): CodeIgniter\HTTP\Exceptions\HTTPException: :forUnkownStatusCode()
#1 /var/www/vhosts/blackserv.de/api.blackserv.de/system/Debug/Exceptions.php(114): CodeIgniter\HTTP\Response-&gt;setStatusCode()
#2 [internal function]: CodeIgniter\Debug\Exceptions-&gt;exceptionHandler()
#3 {main}
thrown in <b>/var/www/vhosts/blackserv.de/api.blackserv.de/system/HTTP/ResponseTrait.php</b> on line <b>158</b><br/>
{
    "title": "ErrorException",
    "type": "ErrorException",
    "code": 500,
    "message": "Uncaught CodeIgniter\\HTTP\\Exceptions\\HTTPException: Unknown HTTP status code provided with no message: 167 in /var/www/vhosts/blackserv.de/api.blackserv.de/system/HTTP/ResponseTrait.php:158\nStack trace:\n#0 /var/www/vhosts/blackserv.de/api.blackserv.de/system/HTTP/ResponseTrait.php(158): CodeIgniter\\HTTP\\Exceptions\\HTTPException::forUnkownStatusCode()\n#1 /var/www/vhosts/blackserv.de/api.blackserv.de/system/Debug/Exceptions.php(114): CodeIgniter\\HTTP\\Response->setStatusCode()\n#2 [internal function]: CodeIgniter\\Debug\\Exceptions->exceptionHandler()\n#3 {main}\n  thrown",
    "file": "/var/www/vhosts/blackserv.de/api.blackserv.de/system/HTTP/ResponseTrait.php",
    "line": 158,
    "trace": [
        {
            "function": "shutdownHandler",
            "class": "CodeIgniter\\Debug\\Exceptions",
            "type": "->"
        }
    ]
}
```

### Steps to Reproduce

```php
<?php
namespace App\Controllers;
use CodeIgniter\API\ResponseTrait;
use CodeIgniter\Controller;
use App\Libraries\AuthSystem;

class Authentication extends BaseController
{
    use ResponseTrait;

    public function index()
    {
        $data = ['options' => ['1' => '/authentication/login', '2' => '/authentication/register'], 'server' => getEnv("ServerID") , ];
        return $this->respond($data);
    }

    public function login()
    {
        $auth = new AuthSystem();
        if (isset($_POST['username']) || isset($_POST['email']) && isset($_POST['password']))
        {
            if (isset($_POST['username']))
            {
                $login = $auth->login($_POST['username'], $_POST['password']);
                $userdat = $auth->find($_POST['username']);
            }
            if (isset($_POST['email']))
            {
                $login = $auth->login($_POST['email'], $_POST['password']);
                $userdat = $auth->find($_POST['email']);
            }
            if ($login == "success")
            {	
				
                $sessiontoken = $auth->regentoken($userdat->id, $this->request->getUserAgent(), $this->request->getIPAddress());

                $data = ['state' => 'succes', 'session' => $sessiontoken, 'server' => getEnv("ServerID") , ];
                return $this->respond($data);

            }
            else
            {
                $data = ['state' => 'error', 'message' => $login, 'server' => getEnv("ServerID") , ];
				return $this->respond($data);
            }

        }
        else
        {
            $data = ['state' => 'error', 'message' => 'not all parameters given', 'server' => getEnv("ServerID") , ];
            return $this->respond($data);
        }
		return $this->respond($data);
    }

}
```

### Expected Output

nothing

### Anything else?

PLEASE HELP

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes  #6227

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Debug/Exceptions.php [hunk 1]
```diff
@@ -115,7 +116,14 @@ public function exceptionHandler(Throwable $exception)
         }
 
         if (! is_cli()) {
-            $this->response->setStatusCode($statusCode);
+            try {
+                $this->response->setStatusCode($statusCode);
+            } catch (HTTPException $e) {
+                // Workaround for invalid HTTP status code.
+                $statusCoden = 500;
+                $this->response->setStatusCode($statusCoden);
```
Comment (paulbalandan) [inline system/Debug/Exceptions.php:124]: Is `$statusCoden` intentional?

Comment (kenjis) [reply system/Debug/Exceptions.php]: It is just a typo.
Fixed.

**Review by MGatner — APPROVED** [review]
Definitely much narrower scope but it gets the job done.

**Review by paulbalandan — APPROVED** [review]

**Comment by kenjis** [comment]
Thank you, guys!

**Pull request merged**